test/tbbmalloc/test_malloc_compliance.cpp

```diff
@@ -1003,7 +1003,7 @@ void CMemTest::RunAllTests(int total_threads)
     TestAlignedParameters();
     UniquePointer();
     AddrArifm();
-#if __APPLE__ || __TBB_USE_THREAD_SANITIZER
+#if __APPLE__ && __HAIKU__ || __TBB_USE_THREAD_SANITIZER
```

Comment:
It should be 'or' condition.
```suggestion
#if __APPLE__ || __HAIKU__ || __TBB_USE_THREAD_SANITIZER
```